optimum/fx/optimization/transformations.py

```diff
@@ -463,6 +464,292 @@ def reverse(self, graph_module: "GraphModule") -> "GraphModule":
+    inp = {"pixel_values": torch.rand(8, 3, 224, 224)}
+
+    traced_model = symbolic_trace(
+        model,
+        input_names=["pixel_values"],
+        disable_check=True
+    )
+
+    non_transformed_time = benchmark(traced_model, inp)
+
+    transformation = FuseBatchNorm2dInConv2d()
+    transformed_model = transformation(traced_model)
+
+    transformed_time = benchmark(transformed_model, inp)
+    print(f"Non-transformed: {non_transformed_time:.2f} s")
+    print(f"Transformed: {transformed_time:.2f} s")
+    print(f"Gain: {(non_transformed_time - transformed_time) / non_transformed_time * 100:.2f} % in latency")
+    ```
+
+    Outputs:
+    ```
+    Non-transformed: 5.06 s
+    Transformed: 4.57 s
+    Gain: 9.71 % in latency
+    ```
+
+    Note that the gain may depend on the batch size, image size, hardware used.
+    """
+
+    preserves_computation = True
+
+    def transform(self, graph_module: "GraphModule") -> "GraphModule":
+        modules = dict(graph_module.named_modules())
+
+        for node in graph_module.graph.nodes:
+            if node.op == "call_module" and node.args[0].op == "call_module":
+                if (
+                    type(modules[node.target]) is torch.nn.BatchNorm2d
+                    and type(modules[node.args[0].target]) is torch.nn.Conv2d
+                ):
+                    if len(node.args[0].users) > 1:  # Output of conv is used by other nodes
+                        continue
+
+                    fused_conv = self.fuse(conv2d=modules[node.args[0].target], bn2d=modules[node.target])
+
+                    # replace the old nn.Conv2d by the fused one
+                    parent_name, _, name = node.args[0].target.rpartition(".")
+                    setattr(modules[parent_name], name, fused_conv)
+
+                    # delete batchnorm from the modules
+                    parent_name, _, name = node.target.rpartition(".")
+                    delattr(modules[parent_name], name)
+
+                    node.replace_all_uses_with(node.args[0])
+                    graph_module.graph.erase_node(node)
+
+        return graph_module
+
+    def fuse(self, conv2d: torch.nn.Conv2d, bn2d: torch.nn.BatchNorm2d):
+        fused_conv = copy.deepcopy(conv2d)
+
+        # handle the case where there is no bias in the conv or the batchnorm has no learnable parameters
+        conv_b = conv2d.bias if conv2d.bias is not None else torch.zeros_like(bn2d.running_mean)
+        bn_w = bn2d.weight if bn2d.weight is not None else torch.ones_like(bn2d.running_mean)
+        bn_b = bn2d.bias if bn2d.bias is not None else torch.ones_like(bn2d.running_mean)
+
+        bn_var_rsqrt = torch.rsqrt(bn2d.running_var + bn2d.eps)
+
+        fused_conv.weight = torch.nn.Parameter(
+            conv2d.weight * (bn_w * bn_var_rsqrt).reshape([-1] + [1] * (len(conv2d.weight.shape) - 1))
+        )
+
+        fused_conv.bias = torch.nn.Parameter(conv_b - bn2d.running_mean * bn_var_rsqrt * bn_w + bn_b)
```

Comment:
I did not manage to do it inplace in the case where the bias is None originally.

Reply:
Can confirm I did not find a way to do it inplace; maybe related to the fact that the bias may be None originally. Do you have an idea? I tried `conv2d.weight.data.copy_(...)`, `conv2d.weight = torch.nn.Parameter(...)` without success. Otherwise I suggest to leave it as is

Reply:
```python
conv2d.weight = torch.nn.Parameter(
            conv2d.weight * (bn_w * bn_var_rsqrt).reshape([-1] + [1] * (len(conv2d.weight.shape) - 1))
        )
```
This does not work? What's the issue?
For instance for `MergeLinears`, I am able to do the following:
```python
merged_linear.weight = torch.nn.Parameter(new_weight)
```